inkstitch.py

```diff
@@ -2,27 +2,101 @@
 #
 # Copyright (c) 2010 Authors
 # Licensed under the GNU GPL version 3.0 or later.  See the file LICENSE for details.
-import cProfile
-import pstats
-import logging
+
 import os
 import sys
+from pathlib import Path
+import configparser
+import lib.debug_utils as debug_utils
+
+SCRIPTDIR = Path(__file__).parent.absolute()
+
+if len(sys.argv) < 2:
+    # no arguments - prevent accidentally running this script
+    print("No arguments given, continue without arguments?")
+    answer = input("Continue? [y/N] ")
+    if answer.lower() != 'y':
+        exit(1)
+
+running_as_frozen = getattr(sys, 'frozen', None) is not None  # check if running from pyinstaller bundle
+
+ini = configparser.ConfigParser()
+ini.read(SCRIPTDIR / "DEVEL.ini")  # read DEVEL.ini file if exists
+
+# prefer pip installed inkex over inkscape bundled inkex, pip version is bundled with Inkstitch
+prefere_pip_inkex = ini.getboolean("LIBRARY","prefer_pip_inkex", fallback=True)  
+
+# check if running from inkscape, given by environment variable
+if os.environ.get('INKSTITCH_OFFLINE_SCRIPT', '').lower() in ['true', '1', 'yes', 'y']:
+    running_from_inkscape = False
+else:
+    running_from_inkscape = True
+
+debug_active = bool((gettrace := getattr(sys, 'gettrace')) and gettrace())  # check if debugger is active on startup
```

Comment:
This is the first use of `:=` in this project.  I am in favor of `:=` in general, if it improves readability, but I think in this case it kind of just complicates this line...